src/OrchardCore.Modules/OrchardCore.Users/Services/DefaultExternalLoginUserToRelateFinder.cs

```diff
@@ -0,0 +1,42 @@
+
+using System.Runtime.CompilerServices;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Identity;
+using OrchardCore.Users.Abstractions;
+
+namespace OrchardCore.Users.Services;
+
+public class DefaultExternalLoginUserToRelateFinder : IExternalLoginUserToRelateFinder
+{
+  private readonly UserManager<IUser> _userManager;
+
+  public DefaultExternalLoginUserToRelateFinder(UserManager<IUser> userManager)
+  {
+    _userManager = userManager;
+  }
+
+  public bool CanManageThis(string extLoginKind)
+  {
+    return true;
+  }
+
+  public async Task<IUser> FindUserToRelateAsync(ExternalLoginInfo info)
+  {
+    //the default behavior previously used in OrchardCore
+    var email = info.Principal.FindFirstValue(ClaimTypes.Email) ?? info.Principal.FindFirstValue("email");
+    
+    IUser iUser = null;
+    if (!string.IsNullOrWhiteSpace(email))
+    {
+      iUser = await _userManager.FindByEmailAsync(email);
+    }
+
+    return iUser;
+  }
+
+  public string GetValueThatLinkAccount(ExternalLoginInfo info)
```

Comment:
GetEmailAync ?

Reply:
the meaning of this changes is to give the opportunity to choose wich parameter use to link the login information to an existing account (or disable this option) ... use "Email" in the method name could be erroneous.